src/main/scala/net/psforever/services/local/support/CaptureFlagManager.scala

```diff
@@ -0,0 +1,282 @@
+  private[this] val log = org.log4s.getLogger(self.path.name)
+
+  var galaxyService: ActorRef = ActorRef.noSender
+
+  private var mapUpdateTick: Cancellable = Default.Cancellable
+
+  /** An internally tracked list of current flags, to avoid querying AmenityOwners each second for flag lookups */
+  private var flags: List[CaptureFlag] = Nil
+
+  private def TrackFlag(flag: CaptureFlag): Unit = {
+    flag.Owner.Amenities = flag
+    flags = flags :+ flag
+
+    if (mapUpdateTick.isCancelled) {
+      // Start sending map updates periodically
+      import scala.concurrent.ExecutionContext.Implicits.global
+      mapUpdateTick = context.system.scheduler.scheduleAtFixedRate(0 seconds, 1 second, self, CaptureFlagManager.MapUpdate())
+    }
+  }
+
+  private def UntrackFlag(flag: CaptureFlag): Unit = {
+    flag.Owner.RemoveAmenity(flag)
+    flags = flags.filterNot(x => x == flag)
+
+    if (flags.isEmpty) {
+      mapUpdateTick.cancel()
+
+      // Send one final map update to clear the last flag from the map
+      self ! CaptureFlagManager.MapUpdate()
+    }
+  }
+
+  val serviceManager = ServiceManager.serviceManager
+  serviceManager ! Lookup("galaxy")
+
+  def receive: Receive = {
+    case LookupResult("galaxy", endpoint) =>
+      galaxyService = endpoint
+
+    case CaptureFlagManager.MapUpdate() =>
+      val flagInfo = flags.map(flag =>
+        FlagInfo(
+          u1 = 0,
+          owner_map_id = flag.Owner.asInstanceOf[Building].MapId,
+          target_map_id = flag.Target.MapId,
+          x = flag.Position.x,
+          y = flag.Position.y,
+          u6 = 0,
+          isMonolithUnit = false
+        )
+      )
+
+      galaxyService ! GalaxyServiceMessage(GalaxyAction.FlagMapUpdate(CaptureFlagUpdateMessage(zone.Number, flagInfo)))
+
+    case CaptureFlagManager.SpawnCaptureFlag(capture_terminal, target, hackingFaction) =>
+      val zone = capture_terminal.Zone
+      val socket = capture_terminal.Owner.asInstanceOf[Building].GetFlagSocket.get
+
+      // Override CC message when looked at
+        zone.LocalEvents ! LocalServiceMessage(
+        zone.id,
+        LocalAction.SendGenericObjectActionMessage(
+          PlanetSideGUID(-1),
+          capture_terminal.GUID,
+          GenericObjectActionEnum.FlagSpawned
+        )
+      )
+
+      // Register LLU object create task and callback to create on clients
+      val flag: CaptureFlag = CaptureFlag.Constructor(
+        Vector3(socket.Position.x, socket.Position.y, socket.Position.z - 1),
+        socket.Orientation,
+        target,
+        socket.Owner,
+        hackingFaction
+      )
+
+      // Add the flag as an amenity and track it internally
+      TrackFlag(flag)
+
+      taskResolver ! CallBackForTask(
+        TaskResolver.GiveTask(GUIDTask.RegisterObjectTask(flag)(socket.Zone.GUID).task),
+        socket.Zone.LocalEvents,
+        LocalServiceMessage(
+          socket.Zone.id,
+          LocalAction.LluSpawned(PlanetSideGUID(-1), flag)
+        )
+      )
```

Comment:
Do LLU's really not have fixed GUID's?  That's crazy.  They're as much an amenity of their facility as their socket is.

Reply:
Nope, at this point we've accounted for pretty much all fixed GUIDs in every zone in the amenity extractor script.